File: commands/readme.js
const chalk = require('chalk');
const { cloneRepo } = require('../lib/git');
const { readFileContent } = require('../lib/utils');
const path = require('path');

async function readme(repo, options) {
  try {
    console.log(chalk.blue(`\n📖 Getting README from ${repo}...\n`));

    // Clone repository
    const repoPath = await cloneRepo(repo, options.cacheDir, options.proxy);

    // Try different README filenames
    const readmeNames = [
      'README.md',
      'readme.md',
      'README.MD',
      'README.markdown',
      'README.txt',
      'README'
    ];

    let readmeContent = null;
    let foundName = null;

    for (const name of readmeNames) {
      const content = readFileContent(path.join(repoPath, name));
      if (content !== null) {
        readmeContent = content;
        foundName = name;
        break;
      }
    }

    // Also check in docs/ folder
    if (!readmeContent) {
      for (const name of readmeNames) {
        const content = readFileContent(path.join(repoPath, 'docs', name));
        if (content !== null) {
          readmeContent = content;
          foundName = `docs/${name}`;
          break;
        }
      }
    }

    if (!readmeContent) {
      console.log(chalk.yellow('No README found.\n'));
      return;
    }

    // Display README
    console.log(chalk.bold(`${foundName}\n`));
    console.log(chalk.gray('─'.repeat(80)));
    console.log(readmeContent);
    console.log(chalk.gray('─'.repeat(80)));

  } catch (error) {
    console.error(chalk.red(`\n❌ Error: ${error.message}`));
    process.exit(1);
  }
}

module.exports = { readme };

File: cmd/readme.go
package cmd

import (
	"fmt"
	"path/filepath"

	"github.com/spf13/cobra"
	"github.com/syxc/ghr/internal/git"
	"github.com/syxc/ghr/internal/utils"
)

// readmeCmd represents the readme command
var readmeCmd = &cobra.Command{
	Use:   "readme <repo>",
	Short: "Get repository README",
	Long:  `Read the README file from a GitHub repository.`,
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		return runReadme(args[0])
	},
}

func init() {
	rootCmd.AddCommand(readmeCmd)
}

func runReadme(repo string) error {
	cfg := getConfig()

	fmt.Println(utils.Blue(fmt.Sprintf("\n📖 Getting README from %s...\n", repo)))

	// Clone repository
	repoPath, err := git.CloneRepo(repo, cfg.CacheDir, cfg.Proxy)
	if err != nil {
		utils.PrintError("%v", err)
		return err
	}

	// Try different README filenames
	readmeNames := []string{
		"README.md",
		"readme.md",
		"README.MD",
		"README.markdown",
		"README.txt",
		"README",
	}

	var readmeContent string
	var foundName string

	for _, name := range readmeNames {
		content, err := utils.ReadFileContent(filepath.Join(repoPath, name))
		if err == nil {
			readmeContent = content
			foundName = name
			break
		}
	}

	// Also check in docs/ folder
	if readmeContent == "" {
		for _, name := range readmeNames {
			content, err := utils.ReadFileContent(filepath.Join(repoPath, "docs", name))
			if err == nil {
				readmeContent = content
				foundName = "docs/" + name
				break
			}
		}
	}

	if readmeContent == "" {
		fmt.Println(utils.Yellow("No README found.\n"))
		return nil
	}

	// Display README
	fmt.Println(utils.Bold(foundName))
	fmt.Println()
	utils.PrintSeparator(80)
	fmt.Println(readmeContent)
	utils.PrintSeparator(80)

	return nil
}
